aiida_common_workflows/cli/utils.py

```diff
@@ -58,3 +58,27 @@ def launch_process(process, daemon, **inputs):
         click.echo('Running a {}...'.format(process_name))
         _, node = launch.run_get_node(process, **inputs)
         echo_process_results(node)
+
+
+def get_code_from_list_or_database(codes, entry_point: str):
+    """Return a code that is configured for the given calculation job entry point.
+
+    First the list of ``Code`` instances is scanned for a suitable code and if not found, one is attempted to be loaded
+    from the database. If no codes are found, ``None`` is returned.
+
+    :param codes: list of aiida `Code` nodes.
+    :param entry_point: calculation job entry point name.
+    :return: a ``Code`` instance configured for the given entry point or ``None``.
+    """
+    from aiida.orm import QueryBuilder, Code
+
+    for entry in codes:
+        if entry.get_attribute('input_plugin') == entry_point:
+            return entry
```

Comment:
I would return here a warning saying that you passed a code that is invalid in case `entry.get_attribute('input_plugin') != entry_point`.
Imagine for instance you pass a code for a different plugin (simple misspell ph instead of pw), no warning is returned and the task is carried on with a random code chosen in the database.

Reply:
That wouldn't make sense with how this function is used. In the CLI, we will get a list of codes from the `-X` option that may be for different code plugins. Take for example fleur, you would get a code for `fleur.fleur` and `fleur.inpgen`. The CLI calls this function passing _both_ codes and for each entry point it will have to pick one out. Raising a warning here would be confusing because it would be the intended behavior.